main.py

```diff
@@ -56,8 +56,7 @@ def main(config):
                 trainer.test_loader is not None
             ), "Test dataset is required for making predictions"
             assert config["checkpoint"]
-            run_dir = config.get("run_dir", "./")
-            results_file = Path(run_dir) / "predictions.json"
```

Comment:
Why did we drop `run_dir`?

Reply:
saving in `run_dir` leads to overriding of prediction files from different runs. Saving it in the results directory makes more sense.

Reply:
Sounds good, good to merge then.

Reply:
Let's hold on merging for a bit, taking care of some nuances when writing to multiple files, etc. We don't want them having to run multiple merging scripts.